src/Generator.Rewrite/Program.cs

```diff
@@ -600,19 +600,33 @@ private static GeneratedVariableIdentifier EmitStringOutParameter(MethodDefiniti
 
             // ptr = Marshal.AllocHGlobal(count + 1);
             var count = GetCountAttribute(parameter);
-            if(count.Count != 0) {
+            if (count == null)
+            {
+                // We need a count attribute so we know what size to make the
+                // string buffer. Currently every string out parameter has a
+                // count attribute but this check is in place to make things
+                // clearer if this case is ever hit.
+                throw new Exception(string.Format("{0}({1}) doesn't have a count attribute", method.Name, parameter.Name));
```

Comment:
Let's not throw generic exceptions. This should be an ArgumentException or a new custom exception which is more applicable (AttributeNotFoundException or TypeNotFoundException? I have the latter already implemented in the rewriter in another working branch)